src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```

Comment:
I'd suggest using private instead of protected to follow the codestyle more closely.
```suggestion
    private LocalDate by;
```